weather/www/js/services.js

```diff
@@ -0,0 +1,214 @@
+            var temp = this.roundTemp(
+						weatherStore.current.daily.data[4].temperatureMax);
+			var precip = weatherStore.current.daily.data[4].precipIntensity;
+			return [precip, temp];
+        };
+		
+		// Return weather data for day 5
+        this.weatherDay5 = function() {
+            var temp = this.roundTemp(
+						weatherStore.current.daily.data[5].temperatureMax);
+			var precip = weatherStore.current.daily.data[5].precipIntensity;
+			return [precip, temp];
+        };
+		
+		// Return weather data for day 6
+        this.weatherDay6 = function() {
+            var temp = this.roundTemp(
+						weatherStore.current.daily.data[6].temperatureMax);
+			var precip = weatherStore.current.daily.data[6].precipIntensity;
+			return [precip, temp];
+        };
+		
+		//Return weather for tonight
+        this.tempToMidnightLow = function() {
+            var low = this.tempNow();
+            var start = this.findHourNow();
+            var end = this.findHourMidnight();
+            if (start >= 0 && end >= 0) {
+                for (var i = start; i <= end; i++) {
+                    low = Math.min(low,
+                            weatherStore.current.hourly.data[i].temperature);
+                }
+            }
+
+            return this.roundTemp(low);
+        };
+		
+		//Return the intensity of precipitation
+		this.todayPrecip = function() {
+			return weatherStore.current.hourly.data[0].precipIntensity;
+		};
+		
+		//Return the correct picture for the current day
+		//depending on the current weather
+		this.getPicture = function() {
+			var pic;
+			var precip = weatherStore.current.daily.data[0].precipIntensity;
+			if (precip == 0) {
+				pic = "https://cdn.evbuc.com/eventlogos/29589191/sunnybig.png"
+			};
+			if (precip > 0 && precip <= .1) {
+				pic = "https://cdn3.iconfinder.com/data/icons/weather-icons-1/64/Sun_Behind_Cloud-512.png"
+			};
+			if (precip > .1) {
+				pic = "http://www.iconpng.com/png/pictograms/rain.png"
+			};
+			return pic;
+		};
+		
+		//Return the wind speed for the current day
+		this.todayWind = function() {
+			return weatherStore.current.hourly.data[0].windSpeed;
+		};
+
+        // Return the index into hourly of the hour, if any, which
+        // contains time (unix time in sec).  Return -1 if not found.
+        // Assume the time in hourly.data is the start of the hour.
+        this.findHour = function(time) {
+            var i = 0;
+            while (i < weatherStore.current.hourly.data.length &&
+                   weatherStore.current.hourly.data[i].time > time) {
+                i++;
+            }
+            if (i < weatherStore.current.hourly.data.length) {
+                return i;
+            } else {
+                return -1;
+            }
+        };
+
+        // Return findHour() (i.e., index into hourly) for current time.
+        this.findHourNow = function() {
+            return this.findHour(Date.now() / 1000); // millisec -> sec
```

Comment:
I've seen it suggested to have that 1000 declared as a constant with a name that explains its purpose